File: Beecrowd/1-Iniciante/2715-Dividindo_os_Trabalhos_I-7/2715.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
public class Solution{
    public static List<int> splitString(string a){
        List<int> ans = new List<int>();
        string current = "";
        bool Was = true;
        foreach(char i in a){
            if(i != ' '){
                current+=i;
                Was = false;
            }
            else{
                if(!Was){
                    ans.Add(int.Parse(current));
                    Was = true;
                }
                current="";
            }
        }
        ans.Add(int.Parse(current));
        return ans;
    }
    public static void Main(){
        while(true){
            try{
                int n = int.Parse(Console.ReadLine().Trim());
                List<int> nums = splitString(Console.ReadLine().Trim());
                while(nums.Count < n)nums.AddRange(splitString(Console.ReadLine().Trim()));
                int l =0;
                int r = n-1;
                long Ra = 0;
                long La = 0;
                while(l<r){
                    if(La + nums[l] < Ra+nums[r]){
                        La+=nums[l];
                        l++;
                    }
                    else{
                        Ra+=nums[r];
                        r--;
                    }
                }
                if(Ra < La) Ra += nums[r];
                else La +=nums[l];
                Console.WriteLine(Math.Abs(La-Ra));
                nums.Clear();
            }
            catch(Exception e){
                break;
            }
        }
    }
}

File: Beecrowd/1-Iniciante/2715-Dividindo_os_Trabalhos_I-7/2715.java
import java.util.*;

public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n;
        long[] nums = new long[1000010];
        while(sc.hasNext()){
            n = sc.nextInt();
            for(int i =0;i<n;i++){
                nums[i] = sc.nextInt();
            }
            long Ra = 0, La = 0;
            int r=n-1, l=0;
            while(l<r){
                if(La+nums[l] <= Ra+nums[r]){
                    La+=nums[l];
                    l++;
                }
                else{
                    Ra+=nums[r];
                    r--;
                }
            }
            Ra += (Ra<La)? nums[r] : nums[r]*-1;
            System.out.println(Math.abs(La-Ra));
        }

        sc.close();
    }
}
